packages/client/lib/sync/fullsync.ts

```diff
@@ -182,6 +102,14 @@ export class FullSynchronizer extends Synchronizer {
     const count = height.sub(first).addn(1)
     if (count.lten(0)) return false
 
+    const nextForkBlock = this.config.chainCommon.nextHardforkBlock()
+    if (nextForkBlock) {
+      if (first.gten(nextForkBlock)) {
+        this.config.chainCommon.setHardforkByBlockNumber(first.toNumber())
+        this.hardfork = this.config.chainCommon.hardfork()
+      }
+    }
```

Comment:
👍

Reply:
No, not here since the HF switches on block download and block execution occur independently. A HF change for execution is handled along the [VMExecution.run()](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1068/files#diff-3ff0e2525d3eb1dd38e375f27736fe31ad55e351e512355106ac144739aa6b78R102) method.